fbw-a380x/src/systems/instruments/src/EWD/shared/ArincValueProvider.ts

```diff
@@ -3,27 +3,87 @@
 // SPDX-License-Identifier: GPL-3.0
 
 import { EventBus } from '@microsoft/msfs-sdk';
-import { Arinc429Register } from '@flybywiresim/fbw-sdk';
+import { Arinc429Register, Arinc429Word } from '@flybywiresim/fbw-sdk';
 import { EwdSimvars } from './EwdSimvarPublisher';
 
 export interface Arinc429Values {
   sat: Arinc429Register;
   slatsFlapsStatus: Arinc429Register;
   slatsPosition: Arinc429Register;
   flapsPosition: Arinc429Register;
+  cpiomB1AgsDiscrete: Arinc429Word;
```

Comment:
**suggestion:** use the read-only, barebones variant of this type, `Arinc429WordData`.

Reply:
Probably a noob question, but because this is a discrete word, is there a way to get specific bit value data from `Arinc429WordData`? If I change this here I can no longer use the `.getBitValueOr` method on it, I can just get the raw numeric value.

Reply:
Here's a patch that fixes this. We should really be careful to not add things to concrete classes that can go into the interface in the future.

```patch
Index: fbw-common/src/systems/shared/src/arinc429.ts
<+>UTF-8
===================================================================
diff --git a/fbw-common/src/systems/shared/src/arinc429.ts b/fbw-common/src/systems/shared/src/arinc429.ts
--- a/fbw-common/src/systems/shared/src/arinc429.ts	(revision af049a0ebfb427a4b71eca0562325ff38e76bf42)
+++ b/fbw-common/src/systems/shared/src/arinc429.ts	(date 1726414158716)
@@ -17,6 +17,12 @@
   isFunctionalTest(): boolean;
 
   isNormalOperation(): boolean;
+
+  valueOr(defaultValue: number | undefined | null): number;
+
+  getBitValue(bit: number): boolean;
+
+  getBitValueOr(bit: number, defaultValue: boolean | undefined | null): boolean;
 }
 
 export class Arinc429Word implements Arinc429WordData {
@@ -158,10 +164,20 @@
     return this.isNormalOperation() ? this.value : defaultValue;
   }
 
+  getBitValue(bit: number): boolean {
+    return ((this.value >> (bit - 1)) & 1) !== 0;
+  }
+
+  getBitValueOr(bit: number, defaultValue: boolean | undefined | null): boolean {
+    return this.isNormalOperation() ? ((this.value >> (bit - 1)) & 1) !== 0 : defaultValue;
+  }
+
+  /** @deprecated use {@link getBitValue} */
   bitValue(bit: number): boolean {
     return ((this.value >> (bit - 1)) & 1) !== 0;
   }
 
+  /** @deprecated use {@link getBitValueOr} */
   bitValueOr(bit: number, defaultValue: boolean | undefined | null): boolean {
     return this.isNormalOperation() ? ((this.value >> (bit - 1)) & 1) !== 0 : defaultValue;
   }
@@ -227,6 +243,18 @@
     return this.word.isNormalOperation();
   }
 
+  valueOr(defaultValue: number | undefined | null): number {
+    return this.word.valueOr(defaultValue);
+  }
+
+  getBitValue(bit: number): boolean {
+    return this.word.getBitValue(bit);
+  }
+
+  getBitValueOr(bit: number, defaultValue: boolean | undefined | null): boolean {
+    return this.word.getBitValueOr(bit, defaultValue);
+  }
+
   async writeToSimVarIfDirty() {
     if (this.isDirty) {
       this.isDirty = false;
```

Reply:
Thanks this worked.